### PR TITLE
Fix `no std` by moving `get_random` dev-deps

### LINKED ISSUE #111: no_std Support
Related

- ~~https://github.com/dalek-cryptography/x25519-dalek/issues/88~~ - Red herring - not this crate

## rand_core?/getrandom

It seems getrandom is brought by rand_core which isn't optional - that stops thumb building

```
    --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/getrandom-0.2.8/src/lib.rs:268:9
    |
268 | /         compile_error!("target is not supported, for more information see: \
269 | |                         https://docs.rs/getrandom/#unsupported-targets");
```

https://github.com/pinkforest/x25519-dalek/actions/runs/4395315721/jobs/7697018808

Maybe we can make the get_random weak and gate via getrandom feature ?

This ofc requires some work around API re: providing randomness

### PR DESCRIPTION
**Pull request by pinkforest**
Fixes #111 

Move the `get_random` weak feature under `rand_core` as dev-dependency so we don't bring std that it brings.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -55,3 +55,5 @@ serde = ["dep:serde", "curve25519-dalek/serde"]
 alloc = ["curve25519-dalek/alloc", "serde?/alloc", "zeroize?/alloc"]
 precomputed-tables = ["curve25519-dalek/precomputed-tables"]
 reusable_secrets = []
+# docs: docsrs and doctest features
+docsrs = ["rand_core/getrandom"]
```
Comment (tarcieri) [inline Cargo.toml:59]: curve25519-dalek seems to solve this by enabling the `getrandom` feature under `dev-dependencies`.

As far as an actual feature goes, I'd suggest calling this one `getrandom`

Comment (pinkforest) [reply Cargo.toml]: Oh wait yeah doctests ran with `dev-dependencies` - oh yeah it was just bench that needed it oops!
Have replaced with `dev-dependencies` to align with curve :+1:

Comment (pinkforest) [reply Cargo.toml]: Also there is getrandom feature now as well which provides the convenience functions which have their own tests
- https://github.com/dalek-cryptography/x25519-dalek/pull/113

**Comment by pinkforest** [comment]
@tarcieri @rozbb this also got closed because the branch was deleted. And can't be re-opened ?

**Pull request closed without merging**